Apps/Database/src/Delegates/DBCommunicationEmailDelegate.cs

```diff
@@ -0,0 +1,93 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//-------------------------------------------------------------------------
+namespace HealthGateway.Database.Delegates
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+    using System.Text.Json;
+    using HealthGateway.Database.Constants;
+    using HealthGateway.Database.Context;
+    using HealthGateway.Database.Models;
+    using HealthGateway.Database.Wrapper;
+    using Microsoft.EntityFrameworkCore;
+    using Microsoft.EntityFrameworkCore.Internal;
+    using Microsoft.Extensions.Logging;
+
+    /// <inheritdoc />
+    public class DBCommunicationEmailDelegate : ICommunicationEmailDelegate
+    {
+        private readonly ILogger<DBNoteDelegate> logger;
+        private readonly GatewayDbContext dbContext;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DBCommunicationEmailDelegate"/> class.
+        /// </summary>
+        /// <param name="logger">Injected Logger Provider.</param>
+        /// <param name="dbContext">The context to be used when accessing the database.</param>
+        public DBCommunicationEmailDelegate(
+            ILogger<DBNoteDelegate> logger,
+            GatewayDbContext dbContext)
+        {
+            this.logger = logger;
+            this.dbContext = dbContext;
+        }
+
+        /// <inheritdoc />
+        public DBResult<CommunicationEmail> Add(CommunicationEmail communicationEmail, bool commit = true)
+        {
+            this.logger.LogTrace($"Adding Communication Email to DB...");
+            DBResult<CommunicationEmail> result = new DBResult<CommunicationEmail>()
+            {
+                Payload = communicationEmail,
+                Status = DBStatusCode.Deferred,
+            };
+            this.dbContext.CommunicationEmail.Add(communicationEmail);
+            if (commit)
+            {
+                try
+                {
+                    this.dbContext.SaveChanges();
+                    result.Status = DBStatusCode.Created;
+                }
+                catch (DbUpdateException e)
+                {
+                    this.logger.LogError($"Unable to save Communication Email to DB {e}");
+                    result.Status = DBStatusCode.Error;
+                    result.Message = e.Message;
+                }
+            }
+
+            this.logger.LogDebug($"Finished adding Communication in DB");
+            return result;
+        }
+
+        /// <inheritdoc />
+        public List<UserProfile> GetActiveUserProfilesByCommunicationId(Guid communicationId, int maxRows)
+        {
+            this.logger.LogTrace($"Getting active user profiles to send emails from DB...");
+
+            var userProfiles = (from profile in this.dbContext.UserProfile
```

Comment:
I don’t like this as it’s the only location where we use this syntax. Did the query I supply as a base not work?

Reply:
I have a hard time to make it work. I can look further into the conversion of the LINQ to Lambda if needed.

Reply:
@TiagoGraf89  has found a winning lambda to replace this Linq.